File: src/building/transform/slot.ts
import * as ItemTypes from "../../api/item.js";
import { DATA as POB_DATA } from "../../data/pob/index.js";

/**
 * 物品信息在POB中的插槽名称。
 */
export function getSlotName(itemData: ItemTypes.Item): string | undefined {
    const inventoryId = itemData.inventoryId;

    if (inventoryId) {
        if (inventoryId === "Flask") {
            return `Flask ${itemData.x! + 1}`;
        }

        return POB_DATA.slotMap[inventoryId];
    }

    return;
}

File: go/building/transform/slot.go
package transform

import (
	"fmt"

	"github.com/cn-poe-community/cn-poe-utils/go/api"
	"github.com/cn-poe-community/cn-poe-utils/go/data/pob"
)

// GetSlotName 获取物品信息在POB中的插槽名称
func GetSlotName(itemData *api.Item) (string, error) {
	inventoryId := itemData.InventoryId

	if inventoryId != nil {
		if *inventoryId == "Flask" {
			return fmt.Sprintf("Flask %d", *itemData.X+1), nil
		}

		if slotName, ok := pob.DefaultData.SlotMap[*inventoryId]; ok {
			return slotName, nil
		}
	}

	return "", fmt.Errorf("unknown inventoryId: %v", inventoryId)
}
